### PR TITLE
PPLUS-853 Evaluator adjustments

### PR DESCRIPTION
**Pull request by kalaputsik**
Overview

Ticket: https://spryker.atlassian.net/browse/PPLUS-853

Change log:
- Adjusted DB column and Transfer property checks.
- Added Spryker/Shared/Twig/TwigFunctionProvider to ignore list.
- Added PluginFilter.
- Adjusted filters and removed the FIlters interface.

### CODE REVIEW

File: src/Evaluate/Infrastructure/Configuration/ConfigurationProvider.php [hunk 1]
```diff
@@ -59,7 +59,7 @@ public function getCoreDirectory(): array
         return [
             $this->getVendorDirectory() . 'spryker' . DIRECTORY_SEPARATOR,
             $this->getVendorDirectory() . 'spryker-eco' . DIRECTORY_SEPARATOR,
-            $this->getVendorDirectory() . 'spryker-shop' . DIRECTORY_SEPARATOR,
+//            $this->getVendorDirectory() . 'spryker-shop' . DIRECTORY_SEPARATOR,
```
Comment (VladislavStrelchenko) [inline src/Evaluate/Infrastructure/Configuration/ConfigurationProvider.php:62]: We need to found a way how to use spryker-shop for projects that need it.

File: src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php [hunk 2]
```diff
@@ -68,6 +68,7 @@ public function getIgnoreSuffixes(): array
     {
         return [
             '/\/Kernel\//',
+            '/\/Shared\//',
```
Comment (alexanderM91) [inline src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php:71]: Can you please remind why we added `Shared` into the ignore list?

File: src/CodeCompliance/Domain/Checks/NotUnique/Constant.php [hunk 3]
```diff
@@ -54,4 +54,15 @@ public function getViolations(): array
...
         return $violations;
     }
+
+    /**
+     * @param string $value
+     * @param string $projectPrefix
+     *
+     * @return bool
+     */
+    protected function hasProjectPrefix(string $value, string $projectPrefix): bool
```
Comment (alexanderM91) [inline src/CodeCompliance/Domain/Checks/NotUnique/Constant.php:64]: I think you can reuse an existing method in the base class.

File: src/CodeCompliance/Domain/Checks/NotUnique/Method.php [hunk 4]
```diff
@@ -71,6 +72,23 @@ public function getViolations(): array
         return $violations;
     }
 
+    /**
+     * @param \ReflectionClass $class
+     *
+     * @return bool
+     */
+    protected function isPlugin(ReflectionClass $class): bool
```
Comment (alexanderM91) [inline src/CodeCompliance/Domain/Checks/NotUnique/Method.php:80]: Can we create a plugin filter which can be reused in the further checks?

File: src/CodeCompliance/Domain/Checks/NotUnique/Method.php [hunk 5]
```diff
@@ -124,4 +142,15 @@ protected function isTraitMethod(string $method, ReflectionClass $class): bool
 
         return count($traits) > 0;
     }
+
```
Comment (alexanderM91) [inline src/CodeCompliance/Domain/Checks/NotUnique/Method.php:145]: The same, you can reuse this functionality from the base class.

**Review by alexanderM91 — CHANGES_REQUESTED** [review]

File: src/Evaluate/Infrastructure/Configuration/ConfigurationProvider.php (hunk 1 above)
Comment (VladislavStrelchenko) [reply src/Evaluate/Infrastructure/Configuration/ConfigurationProvider.php]: Fixed

File: src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php (hunk 2 above)
Comment (VladislavStrelchenko) [reply src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php:71]: For ignore TwigFunctionProvider, such as Spryker\Shared\Twig\TwigFunctionProvider

File: src/CodeCompliance/Domain/Checks/NotUnique/Constant.php (hunk 3 above)
Comment (VladislavStrelchenko) [reply src/CodeCompliance/Domain/Checks/NotUnique/Constant.php:64]: Method was removed from current class

File: src/CodeCompliance/Domain/Checks/NotUnique/Method.php (hunk 4 above)
Comment (VladislavStrelchenko) [reply src/CodeCompliance/Domain/Checks/NotUnique/Method.php]: Added PluginFilter

File: src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php (hunk 2 above)
Comment (alexanderM91) [reply src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php:71]: Can we filter for the specific class but not the whole namespace.

Comment (VladislavStrelchenko) [reply src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php:71]: Changed to dedicated class '/^Spryker\/Shared\/Twig\/Twig\/TwigFunctionProvider$/',

Comment (alexanderM91) [reply src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php:71]: In this case should we remove `Shared`?

**Comment by alexanderM91** [comment]
Hi @VladislavStrelchenko, please add PR description with the relevant ticket and change log.

File: src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php (hunk 2 above)
Comment (VladislavStrelchenko) [reply src/CodeCompliance/Domain/Checks/Filters/IgnoreListFilter.php]: Thanks, Shared is removed  https://github.com/spryker-sdk/upgrader/pull/18/commits/54d41640341f76b97d90d9a4e5f903e0eaacda8a

**Review by alexanderM91 — APPROVED** [review]

**Pull request merged**